.pre-commit-config.yaml

```diff
@@ -0,0 +1,24 @@
+repos:
+  - repo: local
+    hooks:
+      - id: web-js-lint-check
+        name: Web JS Lint Check
+        entry: bash -c 'make -j$(nproc) -C webui/react check-eslint check-prettier-js'
```

Comment:
Added `nproc` to make pre-commit check faster.
On my mac (10 core CPU), 12s without `-j$(nproc)` option. 8s with `-j$(nproc)` option.
I saw an article that says j option with `nproc + 1` is the fastest but we dont need too much optimization and it was only 0.2s faster

Reply:
The n+1 is faster because it basically keeps a job in the queue. That's from the days when 2-4 CPUs was common, so with modern 10+ core systems, it might make sense to be more... Or maybe less with super fast disks. 🤷‍♂️

All that to say I'd support just leaving it at nprocs.

Reply:
make only runs two jobs here, `check-eslint` and `check-prettier-js` which have no deps. not much to parallelize. I suspect nproc > 2 doesn't do anything.

Reply:
I think so too, but the reason why I put `nproc` instead of a constant number is that if we add more commands in the future, we need to increment the number but I think we might forget to do it. 
I don't mind to switch to a constant number tho. Should I leave it or change it?

Reply:
keep nproc, it is good, I'd even support `nproc + 1` because it's canonical. 
I'm just pointing this out to highlight that benchmarking nproc=10 against nproc=11 may not be meaningful.